src/Controls/src/Core/IndicatorView.cs

```diff
@@ -107,25 +110,51 @@ public IEnumerable ItemsSource
 			set => SetValue(ItemsSourceProperty, value);
 		}
 
-
 		protected override SizeRequest OnMeasure(double widthConstraint, double heightConstraint)
 		{
 			var baseRequest = base.OnMeasure(widthConstraint, heightConstraint);
 
 			if (IndicatorTemplate != null)
 				return baseRequest;
 
-			var defaultSize = IndicatorSize + DefaultPadding + DefaultPadding + 1;
-			var items = Count;
-			var sizeRequest = new SizeRequest(new Size(items * defaultSize, IndicatorSize), new Size(10, 10));
+			var padding = DefaultPadding;
+#if __IOS__
+			padding += 7;
+#endif
+			var defaultSize = IndicatorSize + padding + padding;
+			var items = GetMaximumVisible();
+
+			var sizeRequest = new SizeRequest(new Size(items * defaultSize, IndicatorSize), new Size(DefaultViewSize, DefaultViewSize));
+
 			return sizeRequest;
 		}
 
+		protected override Size MeasureOverride(double widthConstraint, double heightConstraint)
+		{
+			var margin = Margin;
+
+			// Adjust the constraints to account for the margins
+			widthConstraint -= margin.HorizontalThickness;
+			heightConstraint -= margin.VerticalThickness;
+
+			// Use the old measurement override to figure out the xplat size
+			var measure = OnMeasure(widthConstraint, heightConstraint).Request;
```

Comment:
@hartez should this be casting to `IView` and using that?

Reply:
Fixed removed this, and figure why sizing wasn't working